### PR TITLE
Setup DemoApp for XCUITest

### PR DESCRIPTION
**Pull request by tianfeng92**
* Add `DemoApp` as the test project
* Update Github actions to build and run XCUITest test
* Update Circle CI to build and run XCUITest test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,9 +28,11 @@ Simply check out this repo and run the appropriate command below :rocket:
 ### In Sauce Cloud
 
 ```shell
-saucectl run
+saucectl run --env APP=<your-app-ipa-file> --env TEST_APP=<your-test-app-ipa-file>
 ```
 
+:bulb: Click [here](.github/workflows/test.yml) to check more details for how to build your XCUITest test bundle.
```
Comment (alexplischke) [inline README.md]:
```suggestion
:bulb: Click [here](.github/workflows/test.yml) for more details on how to build your XCUITest test bundle.
```

File: README.md [hunk 2]
```diff
@@ -28,9 +28,11 @@ Simply check out this repo and run the appropriate command below :rocket:
 ### In Sauce Cloud
 
 ```shell
-saucectl run
+saucectl run --env APP=<your-app-ipa-file> --env TEST_APP=<your-test-app-ipa-file>
```
Comment (alexplischke) [inline README.md]: The `--env` flag is meant for specifying environment variables on the target device.

**Comment by alexplischke** [comment]
The README needs additional touches. Especially this section:
```
[...]
## Running The Examples
Simply check out this repo and run the appropriate command below :rocket:
### In Sauce Cloud
[...]
```

Since the prebuilt .ipa files are gone, you can't just straight up clone and run this example. The app needs to be built. The README should indicate what steps are needed to get there 🙏

File: .sauce/config.yml [hunk 3]
```diff
@@ -11,13 +11,27 @@ sauce:
...
 
 xcuitest:
-  app: ./apps/SauceLabs-Demo-App.ipa
-  testApp: ./apps/SauceLabs-Demo-App-Runner.XCUITest.ipa
+  app: $APP
+  testApp: $TEST_APP
 
 suites:
   - name: "saucy xcuitest"
     testOptions:
```
Comment (alexplischke) [inline .sauce/config.yml]: I'd recommend removing this line if it's empty.

Comment (tianfeng92) [reply .sauce/config.yml]: removed.

File: README.md [hunk 4]
```diff
@@ -23,14 +23,16 @@ saucectl configure
 
 ## Running The Examples
 
-Simply check out this repo and run the appropriate command below :rocket:
+Simply check out this repo, set your XCUITest test bundles and run the appropriate command below :rocket:
```
Comment (alexplischke) [inline README.md:27]: Just putting the statement a bit closer to where you most likely will need it. The moment the user reads "set your XCUITest test bundles", they'll immediately ponder how to get to those, if all they've been using is the Xcode IDE without every inspecting where the build files end up. For those too impatient to read all of the instructions :D 

```suggestion
:bulb: We also provide [DemoApp](DemoApp/) to demonstrate the test workflow. Click [here](.github/workflows/test.yml) for more details on how to build your XCUITest test bundle.
```

File: README.md [hunk 5]
```diff
@@ -23,14 +23,16 @@ saucectl configure
...
 
 ### In Sauce Cloud
 
 ```shell
-saucectl run
+APP=<your-app-ipa-file> TEST_APP=<your-test-app-ipa-file> saucectl run
 ```
 
+:bulb: We also provide [DemoApp](DemoApp/) to demonstrate the test workflow. Click [here](.github/workflows/test.yml) for more details on how to build your XCUITest test bundle.
+
```
Comment (alexplischke) [inline README.md]:
```suggestion
```

**Review by alexplischke — APPROVED** [review]

**Pull request merged**